tests/unit/linalg/test_matrix_dense.cpp

```diff
@@ -11,6 +11,7 @@
 
 #include "mfem.hpp"
 #include "catch.hpp"
+#include "../../../linalg/dtensor.hpp"
```

Comment:
This may cause issues with out-of-source builds. It'll be good to test it.

Reply:
Just pushed a clean up pass. Unrelated to this PR I did hit the following errors
```
lu-fact/tests/unit/fem/test_pa_kernels.cpp:453: FAILED:
due to unexpected exception with message:
  basic_string::_M_replace_aux
```
I think the complete paths to meshes might be needed?

Reply:
Was this with a CMake build or out-of-source GNU make build? In both cases the `data` directory should be copied or linked (from the source to the build directory), so I'm not sure what the issue is.

Reply:
It was with a cmake build.

Reply:
Hmm, with cmake, I'm getting an error in a different place: segfault at `tests/unit/linalg/test_matrix_rectangular.cpp:165`.

Reply:
The failure at `tests/unit/linalg/test_matrix_rectangular.cpp:165` seems to be coming from the new version of hypre I was using. I posted a note about it at #1422.

Reply:
@artv3, you were right, in the cmake build, the `unit_tests` need the top-level source `data` directory to be copied to the build directory. I just pushed a small fix for that.